File: mod01-aws_interaction/demo-sdk/javascript/s3.js
import { S3Client, ListBucketsCommand, GetObjectCommand, NotFound } from "@aws-sdk/client-s3";
import { loadSharedConfigFiles} from '@aws-sdk/shared-ini-file-loader' 

const REGION = (await loadSharedConfigFiles()).region
const s3Client = new S3Client({ region: REGION });

const list = async () => {
    try {
      const bucket_list = await s3Client.send(new ListBucketsCommand({}));
      console.log("[+] List bucket Success : ", bucket_list.Buckets);

      const bucket = bucket_list.Buckets.filter(b => b.Name.includes("demo-public"))[0]
      if (bucket){
        console.log(`[+] Target Bucket: ${bucket.Name}`)
      }
      else{
        throw new NotFound();
      }
      const file = "translated.txt"
      
      const bucketParams = {
        Bucket: bucket.Name,
        Key: file,
      };
      
      console.log(`[+] Trying to download file "${file}"`)
      
      const obj = await s3Client.send(new GetObjectCommand(bucketParams));
      
      const strFromObj = await obj.Body.transformToString();

      console.log(`[+] Dowloaded object content : "${strFromObj}"`)

    } catch (err) {
      if (err instanceof NotFound){
        console.log("[X] Target Bucket not found... Did you run ../demo-cli/s3.sh ?")
      }
      else{
        console.log("[X] Error", err);
      }
    }
};

await list()
 

File: mod01-aws_interaction/demo-sdk/python/s3.py
import boto3

aws_region = boto3.session.Session().region_name

text = "Hi! I'm your trainer today, nice to meet you. Let's learn some AWS!"
print('[+] InputText: ' + text)

translate = boto3.client(service_name='translate', region_name=aws_region, use_ssl=True)

result = translate.translate_text(Text=text, 
            SourceLanguageCode="en", TargetLanguageCode="de")
print('--- SourceLanguageCode: ' + result.get('SourceLanguageCode'))
print('--- TargetLanguageCode: ' + result.get('TargetLanguageCode'))
print('[+] TranslatedText: ' + result.get('TranslatedText'))


print("[+] Ready to upload to s3")

bucket = ""

#Find Bucket Name
s3_client = boto3.client("s3")
bucket_list = s3_client.list_buckets()
for b in bucket_list['Buckets']:
    if "demo-public" in b["Name"]:
        bucket= b["Name"]
        print('--- Target backet: ' + bucket)

if not bucket:
    raise SystemExit("[X] Bucket not found... Did you run ../demo-cli/s3.sh ?")

body = result.get('TranslatedText').encode('utf8')
s3 = boto3.resource('s3')
obj = s3.Object(bucket, 'translated.txt')
obj.put(Body=body)

print(f'[+] Object uploaded at s3://{obj.bucket_name}/{obj.key}')